controllers/front/listing/CategoryController.php

```diff
@@ -154,20 +154,6 @@ public function initContent()
         }
     }
 
-    /**
-     * overrides layout if category is not visible.
-     *
-     * @return bool|string
-     */
-    public function getLayout()
```

Comment:
Hi @jf-viguier 

I don't think the method should be completely removed. As you said the bug occurs for themes that don't have this template but it doesn't mean it's not useful for classic theme Or any other theme that has this file

Reply:
So instead of completely remove this it should be made more resilient by:
- checking if the template file exists before using it
- or check from the theme layouts

Reply:
```php
        $theme = $this->context->shop->theme;
        $layouts = $theme->getAvailableLayouts();
        if (array_key_exists('layout-full-width', $layouts) && !$this->category->checkAccess($this->context->customer->id) || $this->notFound) {
            return 'layouts/layout-full-width.tpl';
        }
```

Reply:
Maybe we could even add a `hasLayout` method to the Theme class

Reply:
@jolelievre I guess that the getLayout() method is not useful anymore. I've checked without it with an nonexistent category and the default layout is used.

Reply:
@jolelievre I think if theme wants to use a different layout, they can use `theme_settings` for that?
<img width="302" alt="Capture d’écran 2022-04-05 à 10 52 47" src="https://user-images.githubusercontent.com/2168836/161717110-377c73dc-2721-4e5f-bb3c-fbcd84f698dd.png">